backend/app/views.py

```diff
@@ -26,9 +28,31 @@
     MapSquareSerializerWithoutPhotos,
     PhotographerSerializer,
     PhotographerSearchSerializer,
-    CorpusAnalysisResultsSerializer,
+    CorpusAnalysisResultsSerializer
 )
 
+ANALYSIS_TAGS = {
+    'detail_fft2': 'detail_fft2',
+    'find_vanishing_point': 'find_vanishing_point',
+    'foreground_percentage': 'foreground_percentage',
+    'combined_indoor': 'indoor_analysis.combined_indoor',
+    'courtyard_frame': 'indoor_analysis.courtyard_frame',
+    'find_windows': 'indoor_analysis.find_windows',
+    'gradient_analysis': 'indoor_analysis.gradient_analysis',
+    'local_variance': 'local_variance',
+    'mean_detail': 'mean_detail',
+    'photographer_caption_length': 'photographer_caption_length',
+    'resnet18_cosine_similarity': 'photo_similarity.resnet18_cosine_similarity',
+    'resnet18_mean_squares_similarity': 'photo_similarity.resnet18_mean_squares_similarity',
+    'resnet18_pairwise_similarity': 'photo_similarity.resnet18_pairwise_similarity',
+    'pop_density_detection': 'pop_density_detection',
+    'portrait_detection': 'portrait_detection',
+    'stdev': 'stdev',
+    'text_ocr': 'text_ocr',
+    'whitespace_percentage': 'whitespace_percentage',
+    'yolo_model': 'yolo_model'
+}
+
```

Comment:
This constant gave me a bit of pause, since I can see that it's returned in the response to the frontend, and looks to be a straightforward mapping of 'sanitized' analysis names to their module counterparts. Before I started digging into the frontend implementation details, my major concern was that it seemed like the frontend would somehow be handling backend concerns. Looking at how the `analysisTags` property on the response JSON is dissected and used throughout the `Search` and `SearchForm` components, I can see a ton of business logic. 

So, in short: just seeing this made me concerned that the frontend was going to be doing too much work, and though I've only had a few minutes to check out the implementation, I can see that it is indeed doing as lot of work. Would you be able to explain the goal of passing this map to the frontend?

Reply:
I think the only reason I passed the map to the frontend was to convert between the module name and the display name. However, the conversion is only needed when sending the search request to the backend, so I'll only pass the tags to the frontend and map them on the backend.

Reply:
👍